lib/versioning/VersionID.js

```diff
@@ -152,4 +156,94 @@ function decode(str) {
     }
 }
 
-module.exports = { generateVersionId, getInfVid, encode, decode };
+/* tiny version Ids constants:
+ *
+ * Note: base62Integer() cannot encode integers larger than 15 digits
+ * so we assume that TV_TOTAL <= 30 and we cut it in half. Please
+ * revisit if TV_TOTAL is greater than 30.
+ */
+const TV_TOTAL = LENGTH_TS + LENGTH_SEQ;
+const TV_HALF = TV_TOTAL / 2;
+const TV_EPAD = '0'.repeat(Math.ceil(TV_HALF * (Math.log(10) / Math.log(62))));
+const TV_DPAD = '0'.repeat(TV_HALF);
+
+/**
+ * Encode a versionId to obscure internal information contained
+ * in a version ID (less than 40 bytes).
+ *
+ * @param {string} str - the versionId to encode
+ * @return {string} - the encoded tinyVersionId
+ */
+function tinyEncode(str) {
+    const part1 = Number(str.substring(0, TV_HALF));
+    const part2 = Number(str.substring(TV_HALF, TV_TOTAL));
+    const part3 = Buffer.from(str.substring(TV_TOTAL));
+    const enc1 = base62Integer.encode(part1);
+    const enc2 = base62Integer.encode(part2);
+    const enc3 = base62String.encode(part3);
+    return (TV_EPAD + enc1).slice(-TV_EPAD.length) +
+        (TV_EPAD + enc2).slice(-TV_EPAD.length) +
+        enc3;
+}
+
+/**
+ * Decode a tinyVersionId. May return an error if the input string is
+ * invalid hex string or results in an invalid value.
+ *
+ * @param {string} str - the encoded tinyVersionId to decode
+ * @return {(string|Error)} - the decoded versionId or an error
```

Comment:
I am not sure returning a string or error is the best case as handling it is a bit hard.
I think just using @throws is fine?

```javascript
const decoded = tinyDecode('blah');
// have to check type of returned value
if (decoded typeof string) {
   // use decoded?
} else {
    // handle the error?
}
```

Reply:
Note I realize now that it matches the other form so in keeping it consistent I guess it is fine even if not ideal.

Reply:
Yes it is to keep it consistent with the legacy routine.